Pipeline crash on `DB::Exception: Failed to write resp`

```
[BaseDaemon.cpp:368] [\\\"(from thread 971) Terminate called after throwing an instance of DB::Exception\\\
Code: 0, e.displayText() = DB::Exception: Failed to write resp, e.what() = DB::Exception\\\
Stack trace:\\\
\\\
\\\
       0x755875d\\\\tterminate_handler() [tiflash+123045725]\\\
                \\\\tlibs/libdaemon/src/BaseDaemon.cpp:633\\\
  0x7ff19c98ba13\\\\tstd::__terminate(void (*)()) [libc++abi.so.1+236051]\\\
  0x7ff19c98b9b8\\\\tstd::terminate() [libc++abi.so.1+235960]\\\
       0x1c0626b\\\\t__clang_call_terminate [tiflash+29385323]\\\
       0x82e6be4\\\\tDB::PipelineExecutorStatus::consume(DB::ResultHandler&) [tiflash+137260004]\\\
                \\\\tdbms/src/Flash/Executor/PipelineExecutorStatus.cpp:115\\\
       0x82e50b9\\\\tDB::PipelineExecutor::execute(DB::ResultHandler&&) [tiflash+137253049]\\\
                \\\\tdbms/src/Flash/Executor/PipelineExecutor.cpp:92\\\
       0x82e9449\\\\tDB::QueryExecutor::execute(std::__1::function<void (DB::Block const&)>) [tiflash+137270345]\\\
                \\\\tdbms/src/Flash/Executor/QueryExecutor.cpp:27\\\
       0x815a7d3\\\\tDB::DAGDriver<true>::execute() [tiflas\\\"] [source=BaseDaemon] [thread_id=1437]\
\",\"[2023/05/20 00:19:22.664 +08:00] [Error] [BaseDaemon.cpp:368] [\\\"(from thread 956) Terminate called after throwing an instance of DB::Exception\\\
Code: 0, e.displayText() = DB::Exception: Failed to write resp, e.what() = DB::Exception\\\
Stack trace:\\\
\\\
\\\
       0x755875d\\\\tterminate_handler() [tiflash+123045725]\\\
                \\\\tlibs/libdaemon/src/BaseDaemon.cpp:633\\\
  0x7f50b183aa13\\\\tstd::__terminate(void (*)()) [libc++abi.so.1+236051]\\\
  0x7f50b183a9b8\\\\tstd::terminate() [libc++abi.so.1+235960]\\\
       0x1c0626b\\\\t__clang_call_terminate [tiflash+29385323]\\\
       0x82e6be4\\\\tDB::PipelineExecutorStatus::consume(DB::ResultHandler&) [tiflash+137260004]\\\
                \\\\tdbms/src/Flash/Executor/PipelineExecutorStatus.cpp:115\\\
       0x82e50b9\\\\tDB::PipelineExecutor::execute(DB::ResultHandler&&) [tiflash+137253049]\\\
                \\\\tdbms/src/Flash/Executor/PipelineExecutor.cpp:92\\\
       0x82e9449\\\\tDB::QueryExecutor::execute(std::__1::function<void (DB::Block const&)>) [tiflash+137270345]\\\
                \\\\tdbms/src/Flash/Executor/QueryExecutor.cpp:27\\\
       0x815a7d3\\\\tDB::DAGDriver<true>::execute() [tiflas\\\"] [source=BaseDaemon] [thread_id=1602]\
\"]"]
[2023/05/20 00:19:31.883 +08:00] [ERROR] [common_checker.go:137] ["tiflash crash"] [log="[\"[2023/05/20 00:19:22.664 +08:00] [Error] [BaseDaemon.cpp:368] [\\\"(from thread 971) Terminate called after throwing an instance of DB::Exception\\\
Code: 0, e.displayText() = DB::Exception: Failed to write resp, e.what() = DB::Exception\\\
Stack trace:\\\
\\\
\\\
       0x755875d\\\\tterminate_handler() [tiflash+123045725]\\\
                \\\\tlibs/libdaemon/src/BaseDaemon.cpp:633\\\
  0x7ff19c98ba13\\\\tstd::__terminate(void (*)()) [libc++abi.so.1+236051]\\\
  0x7ff19c98b9b8\\\\tstd::terminate() [libc++abi.so.1+235960]\\\
       0x1c0626b\\\\t__clang_call_terminate [tiflash+29385323]\\\
       0x82e6be4\\\\tDB::PipelineExecutorStatus::consume(DB::ResultHandler&) [tiflash+137260004]\\\
                \\\\tdbms/src/Flash/Executor/PipelineExecutorStatus.cpp:115\\\
       0x82e50b9\\\\tDB::PipelineExecutor::execute(DB::ResultHandler&&) [tiflash+137253049]\\\
                \\\\tdbms/src/Flash/Executor/PipelineExecutor.cpp:92\\\
       0x82e9449\\\\tDB::QueryExecutor::execute(std::__1::function<void (DB::Block const&)>) [tiflash+137270345]\\\
                \\\\tdbms/src/Flash/Executor/QueryExecutor.cpp:27\\\
       0x815a7d3\\\\tDB::DAGDriver<true>::execute() [tiflas\\\"] [source=BaseDaemon] [thread_id=1437]\
\",\"[2023/05/20 00:19:22.664 +08:00] [Error] [BaseDaemon.cpp:368] [\\\"(from thread 956) Terminate called after throwing an instance of DB::Exception\\\
Code: 0, e.displayText() = DB::Exception: Failed to write resp, e.what() = DB::Exception\\\
Stack trace:\\\
\\\
\\\
       0x755875d\\\\tterminate_handler() [tiflash+123045725]\\\
                \\\\tlibs/libdaemon/src/BaseDaemon.cpp:633\\\
  0x7f50b183aa13\\\\tstd::__terminate(void (*)()) [libc++abi.so.1+236051]\\\
  0x7f50b183a9b8\\\\tstd::terminate() [libc++abi.so.1+235960]\\\
       0x1c0626b\\\\t__clang_call_terminate [tiflash+29385323]\\\
       0x82e6be4\\\\tDB::PipelineExecutorStatus::consume(DB::ResultHandler&) [tiflash+137260004]\\\
                \\\\tdbms/src/Flash/Executor/PipelineExecutorStatus.cpp:115\\\
       0x82e50b9\\\\tDB::PipelineExecutor::execute(DB::ResultHandler&&) [tiflash+137253049]\\\
                \\\\tdbms/src/Flash/Executor/PipelineExecutor.cpp:92\\\
       0x82e9449\\\\tDB::QueryExecutor::execute(std::__1::function<void (DB::Block const&)>) [tiflash+137270345]\\\
                \\\\tdbms/src/Flash/Executor/QueryExecutor.cpp:27\\\
       0x815a7d3\\\\tDB::DAGDriver<true>::execute() [tiflas\\\"] [source=BaseDaemon] [thread_id=1602]\
\"]"] ["end time"=2023/05/20 00:19:27.243 +08:00] [stack="github.com/pingcap/endless/testcase/tiflash/pkg/checker.TiFlashCrashChecker.func1
\t/home/jenkins/agent/workspace/endless-master-build/testcase/tiflash/pkg/checker/common_checker.go:137"]
```
![fTq7AbKfgf](https://github.com/pingcap/tiflash/assets/24507159/dc810be1-9796-4abd-92cd-a4ffb539ef13)

--- a/dbms/src/Flash/Executor/PipelineExecutorStatus.cpp
+++ b/dbms/src/Flash/Executor/PipelineExecutorStatus.cpp
@@ -16,7 +16,7 @@
 
 namespace DB
 {
-ExecutionResult PipelineExecutorStatus::toExecutionResult() noexcept
+ExecutionResult PipelineExecutorStatus::toExecutionResult()
 {
     std::lock_guard lock(mu);
     return exception_ptr
@@ -24,13 +24,13 @@ ExecutionResult PipelineExecutorStatus::toExecutionResult() noexcept
         : ExecutionResult::success();
 }
 
-std::exception_ptr PipelineExecutorStatus::getExceptionPtr() noexcept
+std::exception_ptr PipelineExecutorStatus::getExceptionPtr()
 {
     std::lock_guard lock(mu);
     return exception_ptr;
 }
 
-String PipelineExecutorStatus::getExceptionMsg() noexcept
+String PipelineExecutorStatus::getExceptionMsg()
 {
     try
     {
@@ -49,13 +49,13 @@ String PipelineExecutorStatus::getExceptionMsg() noexcept
     }
 }
 
-void PipelineExecutorStatus::onErrorOccurred(const String & err_msg) noexcept
+void PipelineExecutorStatus::onErrorOccurred(const String & err_msg)
 {
     DB::Exception e(err_msg);
     onErrorOccurred(std::make_exception_ptr(e));
 }
 
-bool PipelineExecutorStatus::setExceptionPtr(const std::exception_ptr & exception_ptr_) noexcept
+bool PipelineExecutorStatus::setExceptionPtr(const std::exception_ptr & exception_ptr_)
 {
     RUNTIME_ASSERT(exception_ptr_ != nullptr);
     std::lock_guard lock(mu);
@@ -65,12 +65,12 @@ bool PipelineExecutorStatus::setExceptionPtr(const std::exception_ptr & exceptio
     return true;
 }
 
-bool PipelineExecutorStatus::isWaitMode() noexcept
+bool PipelineExecutorStatus::isWaitMode()
 {
     return !result_queue.has_value();
 }
 
-void PipelineExecutorStatus::onErrorOccurred(const std::exception_ptr & exception_ptr_) noexcept
+void PipelineExecutorStatus::onErrorOccurred(const std::exception_ptr & exception_ptr_)
 {
     if (setExceptionPtr(exception_ptr_))
     {
@@ -79,7 +79,7 @@ void PipelineExecutorStatus::onErrorOccurred(const std::exception_ptr & exceptio
     }
 }
 
-void PipelineExecutorStatus::wait() noexcept
+void PipelineExecutorStatus::wait()
 {
     {
         std::unique_lock lock(mu);
@@ -89,7 +89,7 @@ void PipelineExecutorStatus::wait() noexcept
     LOG_DEBUG(log, "query finished and wait done");
 }
 
-ResultQueuePtr PipelineExecutorStatus::getConsumedResultQueue() noexcept
+ResultQueuePtr PipelineExecutorStatus::getConsumedResultQueue()
 {
     std::lock_guard lock(mu);
     RUNTIME_ASSERT(!isWaitMode());
@@ -98,7 +98,7 @@ ResultQueuePtr PipelineExecutorStatus::getConsumedResultQueue() noexcept
     return consumed_result_queue;
 }
 
-void PipelineExecutorStatus::consume(ResultHandler & result_handler) noexcept
+void PipelineExecutorStatus::consume(ResultHandler & result_handler)
 {
     RUNTIME_ASSERT(result_handler);
     auto consumed_result_queue = getConsumedResultQueue();
@@ -114,13 +114,13 @@ void PipelineExecutorStatus::consume(ResultHandler & result_handler) noexcept
     LOG_DEBUG(log, "query finished and consume done");
 }
 
-void PipelineExecutorStatus::onEventSchedule() noexcept
+void PipelineExecutorStatus::onEventSchedule()
 {
     std::lock_guard lock(mu);
     ++active_event_count;
 }
 
-void PipelineExecutorStatus::onEventFinish() noexcept
+void PipelineExecutorStatus::onEventFinish()
 {
     std::lock_guard lock(mu);
     RUNTIME_ASSERT(active_event_count > 0);
@@ -143,12 +143,12 @@ void PipelineExecutorStatus::onEventFinish() noexcept
     }
 }
 
-void PipelineExecutorStatus::cancel() noexcept
+void PipelineExecutorStatus::cancel()
 {
     is_cancelled.store(true, std::memory_order_release);
 }
 
-ResultQueuePtr PipelineExecutorStatus::toConsumeMode(size_t queue_size) noexcept
+ResultQueuePtr PipelineExecutorStatus::toConsumeMode(size_t queue_size)
 {
     std::lock_guard lock(mu);
     RUNTIME_ASSERT(!result_queue.has_value());

--- a/dbms/src/Flash/Executor/PipelineExecutorStatus.h
+++ b/dbms/src/Flash/Executor/PipelineExecutorStatus.h
@@ -39,19 +39,19 @@ public:
         : log(Logger::get(req_id))
     {}
 
-    ExecutionResult toExecutionResult() noexcept;
+    ExecutionResult toExecutionResult();
 
-    std::exception_ptr getExceptionPtr() noexcept;
-    String getExceptionMsg() noexcept;
+    std::exception_ptr getExceptionPtr();
+    String getExceptionMsg();
 
-    void onEventSchedule() noexcept;
+    void onEventSchedule();
 
-    void onEventFinish() noexcept;
+    void onEventFinish();
 
-    void onErrorOccurred(const String & err_msg) noexcept;
-    void onErrorOccurred(const std::exception_ptr & exception_ptr_) noexcept;
+    void onErrorOccurred(const String & err_msg);
+    void onErrorOccurred(const std::exception_ptr & exception_ptr_);
 
-    void wait() noexcept;
+    void wait();
 
     template <typename Duration>
     void waitFor(const Duration & timeout_duration)
@@ -71,7 +71,7 @@ public:
         LOG_DEBUG(log, "query finished and wait done");
     }
 
-    void consume(ResultHandler & result_handler) noexcept;
+    void consume(ResultHandler & result_handler);
 
     template <typename Duration>
     void consumeFor(ResultHandler & result_handler, const Duration & timeout_duration)
@@ -106,14 +106,14 @@ public:
         LOG_DEBUG(log, "query finished and consume done");
     }
 
-    void cancel() noexcept;
+    void cancel();
 
-    ALWAYS_INLINE bool isCancelled() noexcept
+    ALWAYS_INLINE bool isCancelled()
     {
         return is_cancelled.load(std::memory_order_acquire);
     }
 
-    ResultQueuePtr toConsumeMode(size_t queue_size) noexcept;
+    ResultQueuePtr toConsumeMode(size_t queue_size);
 
     void update(const TaskProfileInfo & task_profile_info)
     {
@@ -126,12 +126,12 @@ public:
     }
 
 private:
-    bool setExceptionPtr(const std::exception_ptr & exception_ptr_) noexcept;
+    bool setExceptionPtr(const std::exception_ptr & exception_ptr_);
 
     // Need to be called under lock.
-    bool isWaitMode() noexcept;
+    bool isWaitMode();
 
-    ResultQueuePtr getConsumedResultQueue() noexcept;
+    ResultQueuePtr getConsumedResultQueue();
 
 private:
     LoggerPtr log;

